### PR TITLE
support null and array in type

### PR DESCRIPTION
**Pull request by koxudaxi**
This PR supports null and array in type
## Related Issues
https://github.com/koxudaxi/datamodel-code-generator/issues/223

### CODE REVIEW

**Comment by koxudaxi** [comment]
@JosXa
Would you please review this PR?

File: datamodel_code_generator/parser/jsonschema.py [hunk 1]
```diff
@@ -183,10 +184,21 @@ def get_data_type(self, obj: JsonSchemaObject) -> List[DataType]:
         if isinstance(obj.type, list):
             types: List[str] = [t for t in obj.type if t != 'null']
```
Comment (JosXa) [inline datamodel_code_generator/parser/jsonschema.py:185]: I'm curious if the string `'null'` needs to be stripped here for some weird inputs (e.g. `'  nUlL     '`)

**Review by JosXa — APPROVED** [review]
Cool, thanks a ton!
I can't pretend I understand a lot of what's happening here, but the test case looks good!

File: datamodel_code_generator/parser/jsonschema.py (hunk 1 above)
Comment (koxudaxi) [reply datamodel_code_generator/parser/jsonschema.py:185]: I guess I don't need stripping these inputs.
`null` is defined in the rule book.

**Comment by koxudaxi** [comment]
@JosXa 
Thank you for reviewing the PR.
> I can't pretend I understand a lot of what's happening here, but the test case looks good!
Yes, This parser is a little complex structure to support a few formats in the future.

Of course,  We can refactor the parser.

**Comment by JosXa** [comment]
Might be time.. One thing I noticed is that the code tends to repeat the same name for local vars and self vars and then even other objects with the same names. Think:

```
...
my_var = [x for x in self._my_var]
some_obj.my_var = my_var
```
This made it hard to read. Food for thought :)

**Comment by koxudaxi** [comment]
> Might be time.. One thing I noticed is that the code tends to repeat the same name for local vars and self vars and then even other objects with the same names. Think:

It's my coding style recently...
I agree with your thought. I should change it to read easily. :smiley_cat: 
I want to change all the similar code when I get the chance. 

If you don't have other pointed out then I merge it.

**Comment by JosXa** [comment]
Cool hehe, and glad to be working with a friendly maintainer :) Let's see if I can give this a shot with PRs next time I find a bug 💪😁

LGTM!

**Comment by koxudaxi** [comment]
I'm happy working with you too :smile: 
PRs welcome :rocket:

**Pull request merged**